ambari-server/src/main/java/org/apache/ambari/server/utils/MpackVersion.java

```diff
@@ -0,0 +1,197 @@
+  private final static Pattern PATTERN_WITHOUT_HOTFIX = Pattern.compile(VERSION_WITH_BUILD_PATTERN);
+
+  // Parts of version
+  private int major;
+  private int minor;
+  private int maint;
+  private int hotfix;
+  private int build;
+
+
+  public MpackVersion(int major, int minor, int maint, int hotfix, int build) {
+    this.major = major;
+    this.minor = minor;
+    this.maint = maint;
+    this.hotfix = hotfix;
+    this.build = build;
+  }
+
+  /**
+   * Method which will parse mpack version
+   * which user passed as parameter. Also
+   * in this method version will be validated.
+   * @param mpackVersion string
+   * @return MpackVersion instance which contains parsed version
+   * */
+  public static MpackVersion parse(String mpackVersion) {
+    Matcher versionMatcher = validateMpackVersion(mpackVersion);
+    MpackVersion result = null;
+
+    if (versionMatcher.pattern().pattern().equals(VERSION_WITH_BUILD_PATTERN)) {
+      result = new MpackVersion(Integer.parseInt(versionMatcher.group(1)), Integer.parseInt(versionMatcher.group(2)),
+              Integer.parseInt(versionMatcher.group(3)), 0, Integer.parseInt(versionMatcher.group(4)));
+
+    } else {
+      result = new MpackVersion(Integer.parseInt(versionMatcher.group(1)), Integer.parseInt(versionMatcher.group(2)),
+              Integer.parseInt(versionMatcher.group(3)), Integer.parseInt(versionMatcher.group(4)), Integer.parseInt(versionMatcher.group(5)));
+
+    }
+
+    return result;
+  }
+
+  /**
+   * Method which will parse stack version
+   * which user passed as parameter. Also
+   * in this method version will be validated.
+   * @param stackVersion string
+   * @return MpackVersion instance which contains parsed version
+   * */
+  public static MpackVersion parseStackVersion(String stackVersion) {
+    Matcher versionMatcher = validateStackVersion(stackVersion);
+    MpackVersion result = new MpackVersion(Integer.parseInt(versionMatcher.group(1)), Integer.parseInt(versionMatcher.group(2)),
+          Integer.parseInt(versionMatcher.group(3)), Integer.parseInt(versionMatcher.group(4)), Integer.parseInt(versionMatcher.group(5)));
+
+    return result;
+  }
+
+  /**
+   * Method validate stack version not to be
+   * empty or null. Also check if passed version
+   * has valid format.
+   * @param version string
+   * @return Matcher for passed version
+   * @throws IllegalArgumentException() if version empty/null/not valid
+   */
+  private static Matcher validateStackVersion(String version) {
+    if (StringUtils.isEmpty(version)) {
+      throw new IllegalArgumentException("Stack version can't be empty or null");
+    }
+
+    String stackVersion = StringUtils.trim(version);
+
+    Matcher versionMatcher = PATTERN_WITH_HOTFIX.matcher(stackVersion);
+    if (!versionMatcher.find()) {
+      versionMatcher = PATTERN_LEGACY_STACK_VERSION.matcher(stackVersion);
+      if (!versionMatcher.find()) {
```

Comment:
Hi @vbrodetskyi ,
I guess we need to add another condition for PATTERN_WITHOUT_HOTFIX validation.